internal/cmd/request.go

```diff
@@ -105,13 +107,107 @@ func PrintCMDResponseFull(request any, response core.APIResponse, responseType a
 		fmt.Printf("\nAutograder Response:\n---\n%s\n---\n", util.MustToJSONIndent(response))
 	}
 
+	successfulConversion := false
+	customPrintOutput := ""
 	if customPrintFunc != nil {
-		fmt.Println(customPrintFunc(response))
-	} else if responseType == nil {
-		fmt.Println(util.MustToJSONIndent(response.Content))
+		customPrintOutput, successfulConversion = customPrintFunc(response)
+	}
+
+	output := ""
+	if successfulConversion {
+		output = customPrintOutput
 	} else {
-		responseContent := reflect.New(reflect.TypeOf(responseType)).Interface()
-		util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
-		fmt.Println(util.MustToJSONIndent(responseContent))
+		if responseType != nil {
+			responseContent := reflect.New(reflect.TypeOf(responseType)).Interface()
+			util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
+			output = util.MustToJSONIndent(responseContent)
+		} else {
+			output = util.MustToJSONIndent(response.Content)
+		}
+	}
+
+	fmt.Println(output)
+}
+
+// Attempt to convert an API response into a TSV table.
+// If the response content has multiple keys or a single map, it creates a single row table.
+// If it has one key with a list, the values in each element becomes a row of the table.
+// Return the table and true on a successful conversion or false if there is an issue converting.
+func ConvertApiResponseToTable(response core.APIResponse) (string, bool) {
+	responseContent, ok := response.Content.(map[string]any)
+	if !ok {
+		return "", false
+	}
+
+	if len(responseContent) == 0 {
+		return "", false
+	} else if len(responseContent) > 1 {
+		return convertEntriesToTable([]any{responseContent})
+	} else {
```

Comment:
I would break these up and remove the need for the last else.

I know this may sound confusing, given on how the last pass I told you that mutually exclusive cases should be in the same conditional tree.
But, the difference here is that you handle the zero case and fail fast.
Here is the comment for reference:

> These are mutually exclusive, they should be in the same tree.
> (If they both needed to exist.)
> 
> But, they probably don't all need to exist.
> Fail faster with zero length, then you don't need to have the second conditional.